### PR TITLE
Adding models for event_likes

### PR DESCRIPTION
**Pull request by philipr-za**
Added the model and method to allow a user to like an event and return the total likes per event.

### CODE REVIEW

File: migrations/00000000000014_event_likes/up.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+-- Define the event_likes table
+CREATE TABLE event_likes (
+  id uuid PRIMARY KEY DEFAULT gen_random_uuid() NOT NULL,
+  event_id uuid NOT NULL REFERENCES events (id) ON DELETE CASCADE,
+  user_id uuid NOT NULL REFERENCES users (id) ON DELETE CASCADE
+);
+
+-- Indices
+CREATE INDEX index_event_likes_event_id_user_id ON event_likes (event_id,user_id);
```
Comment (mikethetike) [inline migrations/00000000000014_event_likes/up.sql]: Add `UNIQUE`

File: src/models/event_likes.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        NewEventLike {
+            event_id: event_id,
+            user_id: user_id,
+        }
+    }
+
+    pub fn total_likes(
+        event_id: &Uuid,
+        conn: &Connectable,
+    ) -> Result<EventLikeTotal, DatabaseError> {
```
Comment (mikethetike) [inline src/models/event_likes.rs]: This should return `u32`. I think move the struct `EventLikeTotal` to `bn-api`and use it only for JSON serialization

Comment (mikethetike) [reply src/models/event_likes.rs]: I also think that this should be a method on the `Event` struct. e.g. 
```rust

let e = Event{}.....
e.total_likes();
```

Comment (philipr-za) [reply src/models/event_likes.rs]: All done

**Pull request merged**